### ISSUE TITLE
Unreachable reached when aggregating SentinelVector in DataFrames.jl

### ISSUE THREAD

**Issue by bkamins** [JuliaLang/julia#37179]
As suggested on Slack I split https://github.com/JuliaLang/julia/issues/36923#issuecomment-679300281 here.

The issue is reported here https://github.com/JuliaData/DataFrames.jl/issues/2382. It fails on both Julia 1.5 and 1.6. I am copy-pasting the error here to make the issue stand alone:
```
julia> using DataFrames, CSV, Statistics # DataFrames 0.21.6 or master; CSV.jl - 0.7.7; SentinelArrays.jl v1.2.13

julia> df = CSV.read(IOBuffer("x,y\n1,1.0\n1,\n1,1.5\n2,2.0\n2,\n2,3.0\n"), DataFrame)
6×2 DataFrame
│ Row │ x     │ y        │
│     │ Int64 │ Float64? │
├─────┼───────┼──────────┤
│ 1   │ 1     │ 1.0      │
│ 2   │ 1     │ missing  │
│ 3   │ 1     │ 1.5      │
│ 4   │ 2     │ 2.0      │
│ 5   │ 2     │ missing  │
│ 6   │ 2     │ 3.0      │

julia> typeof.(eachcol(df))
2-element Array{DataType,1}:
 Array{Int64,1}
 SentinelArrays.SentinelArray{Float64,1,Float64,Missing,Array{Float64,1}}

julia> gdf = groupby(df, :x)
GroupedDataFrame with 2 groups based on key: x
First Group (3 rows): x = 1
│ Row │ x     │ y        │
│     │ Int64 │ Float64? │
├─────┼───────┼──────────┤
│ 1   │ 1     │ 1.0      │
│ 2   │ 1     │ missing  │
│ 3   │ 1     │ 1.5      │
⋮
Last Group (3 rows): x = 2
│ Row │ x     │ y        │
│     │ Int64 │ Float64? │
├─────┼───────┼──────────┤
│ 1   │ 2     │ 2.0      │
│ 2   │ 2     │ missing  │
│ 3   │ 2     │ 3.0      │

julia> combine(gdf, :y => var)
Unreachable reached at 0x7fad5ee5da96

signal (4): Illegal instruction
in expression starting at REPL[8]:1
Aggregate at /home/bkamins/.julia/packages/DataFrames/4ov0U/src/groupeddataframe/splitapplycombine.jl:1044
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2231 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2398
_combine at /home/bkamins/.julia/packages/DataFrames/4ov0U/src/groupeddataframe/splitapplycombine.jl:1150
#combine_helper#391 at /home/bkamins/.julia/packages/DataFrames/4ov0U/src/groupeddataframe/splitapplycombine.jl:589
combine_helper##kw at /home/bkamins/.julia/packages/DataFrames/4ov0U/src/groupeddataframe/splitapplycombine.jl:585
unknown function (ip: 0x7fad5ee5d161)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2231 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2398
#_combine_prepare#379 at /home/bkamins/.julia/packages/DataFrames/4ov0U/src/groupeddataframe/splitapplycombine.jl:553
unknown function (ip: 0x7fad5ee506ff)
_combine_prepare##kw at /home/bkamins/.julia/packages/DataFrames/4ov0U/src/groupeddataframe/splitapplycombine.jl:482 [inlined]
#combine#378 at /home/bkamins/.julia/packages/DataFrames/4ov0U/src/groupeddataframe/splitapplycombine.jl:474 [inlined]
combine at /home/bkamins/.julia/packages/DataFrames/4ov0U/src/groupeddataframe/splitapplycombine.jl:474
unknown function (ip: 0x7fad5ee4e225)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2231 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2398
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1690 [inlined]
do_call at /buildworker/worker/package_linux64/build/src/interpreter.c:117
eval_value at /buildworker/worker/package_linux64/build/src/interpreter.c:206
eval_stmt_value at /buildworker/worker/package_linux64/build/src/interpreter.c:157 [inlined]
eval_body at /buildworker/worker/package_linux64/build/src/interpreter.c:566
jl_interpret_toplevel_thunk at /buildworker/worker/package_linux64/build/src/interpreter.c:660
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:840
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:790
jl_toplevel_eval_in at /buildworker/worker/package_linux64/build/src/toplevel.c:883
eval at ./boot.jl:331
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2214 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2398
eval_user_input at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:134
repl_backend_loop at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:195
start_repl_backend at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:180
#run_repl#37 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:292
run_repl at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:288
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2231 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2398
#806 at ./client.jl:399
jfptr_YY.806_52378.clone_1 at /home/bkamins/julia/lib/julia/sys.so (unknown line)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2214 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2398
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1690 [inlined]
do_apply at /buildworker/worker/package_linux64/build/src/builtins.c:655
jl_f__apply_latest at /buildworker/worker/package_linux64/build/src/builtins.c:705
#invokelatest#1 at ./essentials.jl:710 [inlined]
invokelatest at ./essentials.jl:709 [inlined]
run_main_repl at ./client.jl:383
exec_options at ./client.jl:313
_start at ./client.jl:506
jfptr__start_43701.clone_1 at /home/bkamins/julia/lib/julia/sys.so (unknown line)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2214 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2398
unknown function (ip: 0x401931)
unknown function (ip: 0x401533)
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x4015d4)
Allocations: 39121077 (Pool: 39109470; Big: 11607); GC: 41
Illegal instruction (core dumped)
```

**Comment by quinnj** [comment]
This is the same issue (and fix) as https://github.com/JuliaData/DataFrames.jl/issues/2326; I brought it up w/ @Keno at one point and he mentioned it was some kind of function type parameter matching issue in the compiler; and indeed, both fixes have been removing the type parameter from the `AbstractVector` arguments.

**Comment by quinnj** [comment]
(work-around in DataFrames for this issue is https://github.com/JuliaData/DataFrames.jl/pull/2383)

**Comment by vtjnash** [comment]
Fixed by 102c9ac37ddcdb241eb781fea6d168b1823bf5ce
